File: 128. Longest Consecutive Sequence/main.py
from typing import List


class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if len(nums) == 0:
            return 0

        nums.sort()

        count = 1
        maxCount = 1

        for i in range(1, len(nums)):
            if nums[i] == nums[i - 1]:
                continue
            elif nums[i] == nums[i - 1] + 1:
                count += 1
            else:
                count = 1
            maxCount = max(maxCount, count)

        return maxCount


class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if len(nums) == 0:
            return 0

        sets = set(nums)

        maxCount = 1

        for num in sets:
            if num - 1 in sets:
                continue

            count = 1
            last = num

            while last + 1 in sets:
                count += 1
                last += 1

            maxCount = max(maxCount, count)

        return maxCount

File: 128. Longest Consecutive Sequence/main.rs
struct Solution;

impl Solution {
    pub fn longest_consecutive(nums: Vec<i32>) -> i32 {
        if nums.is_empty() {
            return 0;
        }

        let mut nums = nums;
        nums.sort();

        let mut count = 1;
        let mut max_count = 1;

        for i in 1..nums.len() {
            if nums[i] == nums[i - 1] {
                continue;
            } else if nums[i] == nums[i - 1] + 1 {
                count += 1;
            } else {
                count = 1;
            }

            max_count = max_count.max(count);
        }

        max_count
    }
}

use std::collections::HashSet;

impl Solution {
    pub fn longest_consecutive(nums: Vec<i32>) -> i32 {
        if nums.is_empty() {
            return 0;
        }

        let hashset: HashSet<i32> = HashSet::from_iter(nums.into_iter());

        let mut max_count = 1;

        for &num in hashset.iter() {
            if hashset.contains(&(num - 1)) {
                continue;
            }

            let mut count = 1;
            let mut last = num;

            while hashset.contains(&(last + 1)) {
                count += 1;
                last += 1;
            }

            max_count = max_count.max(count);
        }

        max_count
    }
}
